### PR TITLE
Split wallet new static

### PR DESCRIPTION
**Pull request by boothead**
## Description

I'm splitting `cardano-sl-wallet-new` in the top level default.nix. I can't use it as a dependency in its current form as `justStaticExecutables` throws away the library.

I've also pulled in @disassembler's changes ignoring `dist` and other local files. That's a separate commit and can easily be dropped if we'd prefer to wait for his original changes to reach develop.

## Type of change

- [x] 🐞 Bug fix (non-breaking change which fixes an issue)
- [~] 🛠 New feature (non-breaking change which adds functionality)
- [~] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)
- [x] 🏭 Refactoring that does not change existing functionality but does improve things like code readability, structure etc
- [ ] 🔨 New or improved tests for existing code
- [ ] ⛑ git-flow chore (backport, hotfix, etc)

## Developer checklist

- [~] I have read the [style guide](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/style-guide.md) document, and my code follows the code style of this project.
- [~] If my code deals with exceptions, it follows the [guidelines](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/exceptions.md).
- [~] I have updated any documentation accordingly, if needed. Documentation changes can be reflected in opening a PR on [cardanodocs.com](https://github.com/input-output-hk/cardanodocs.com), amending the inline [Haddock](https://www.haskell.org/haddock/) comments, any relevant README file or one of the document listed in the [docs](https://github.com/input-output-hk/cardano-sl/tree/develop/docs) directory.

## Testing checklist

- [~] I have added tests to cover my changes.
- [~] All new and existing tests passed.

## QA Steps

I need some code review on where the new derivation `cardano-sl-wallet-new-static` should be used instead of `cardano-sl-wallet-new`

@adinapoli-iohk @disassembler @jmitchell @cleverca22 @k0001 

Supersedes #2928

### CODE REVIEW

File: default.nix [hunk 1]
```diff
@@ -61,7 +61,8 @@ let
         executableHaskellDepends = drv.executableHaskellDepends ++ [self.cabal-install];
       })));
       cardano-sl-node = addGitRev super.cardano-sl-node;
-      cardano-sl-wallet-new = addGitRev (justStaticExecutables super.cardano-sl-wallet-new);
+      cardano-sl-wallet-new = addGitRev super.cardano-sl-wallet-new;
+      cardano-sl-wallet-new-static = justStaticExecutables cardano-sl-wallet-new;
```
Comment (jmitchell) [inline default.nix]: `error: undefined variable 'cardano-sl-wallet-new' at /var/lib/buildkite-agent/builds/buildkite-agent-4/input-output-hk/cardano-sl/default.nix:65:60`

I believe you're just missing the `super.` here.

Comment (akegalj) [reply default.nix]: yes, seems like extra `super.` here

**Comment by jcmincke** [comment]
Closed because this PR is just too old.

**Comment by disassembler** [comment]
this also was already done in [DEVOPS-1027] make hydra show metrics for derivations  [Samuel Leathers] so perfectly correct to close.

**Pull request closed without merging**